pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -198,10 +199,19 @@ func (cra *cpuResourceAdvisor) GetHeadroom() (resource.Quantity, error) {
 // update works in a monolithic way to maintain lifecycle and triggers update actions for all regions;
 // todo: re-consider whether it's efficient or we should make start individual goroutine for each region
 func (cra *cpuResourceAdvisor) update() {
+	if !cra.updateWithIsolationGuardian(true) {
+		cra.updateWithIsolationGuardian(false)
+	}
+}
+
+// updateWithIsolationGuardian returns true if the process works as expected,
+// otherwise, we should retry with the isolation disabled
+// todo: we should re-design the mechanises of isolation instead of disabling this functionality
+func (cra *cpuResourceAdvisor) updateWithIsolationGuardian(enableIsolated bool) bool {
```

Comment:
better to use tryIsolation instead of enableIsolated?

Reply:
okay